client/extensions/woocommerce/app/products/product-create.js

```diff
@@ -84,32 +86,34 @@ class ProductCreate extends React.Component {
 	}
 
 	render() {
-		const { product, className, variations, productCategories } = this.props;
+		const { siteId, product, className, variations, productCategories } = this.props;
 
 		return (
 			<Main className={ className }>
 				<SidebarNavigation />
 				<ProductHeader
 					onTrash={ this.onTrash }
 					onSave={ this.onSave }
+					saveDisabled={ ! siteId }
 				/>
-				<ProductForm
+				{ siteId && ( <ProductForm
+					siteId={ siteId }
 					product={ product || { type: 'simple' } }
 					variations={ variations }
 					productCategories={ productCategories }
 					editProduct={ this.props.editProduct }
 					editProductAttribute={ this.props.editProductAttribute }
 					editProductVariation={ this.props.editProductVariation }
-				/>
+				/> ) }
 			</Main>
 		);
 	}
 }
 
 function mapStateToProps( state ) {
 	const siteId = getSelectedSiteId( state );
-	const product = getCurrentlyEditingProduct( state );
-	const variations = product && getProductVariationsWithLocalEdits( state, product.id );
+	const product = getCurrentlyEditingProduct( state, siteId );
+	const variations = product && getProductVariationsWithLocalEdits( state, product.id, siteId );
```

Comment:
I think we should keep the order as is. All of our selectors currently put `siteId` at the back (since you can omit it and have them default to current selected site), where as all of our actions put it first.